pkg/minikube/driver/driver.go

```diff
@@ -141,17 +145,69 @@ func IsDocker(name string) bool {
 	return name == Docker
 }
 
-// IsDockerDesktop checks if the driver is a Docker for Desktop (Docker on windows or MacOs)
+func IsLinuxDockerDesktop(name string) bool {
+	if IsDocker(name) {
+		cmd := exec.Command("docker", "version")
+		stdout, err := cmd.StdoutPipe()
+		if err != nil {
+			return false
+		}
+		if err := cmd.Start(); err != nil {
+			return false
+		}
+		data, err := ioutil.ReadAll(stdout)
+		if err != nil {
+			return false
+		}
+		output := string(data)
+		return strings.Contains(output, "Desktop")
+	}
+	return false
+}
+
+// IsDockerDesktop checks if the driver is a Docker for Desktop
 // for linux and exotic archs, this will be false
 func IsDockerDesktop(name string) bool {
 	if IsDocker(name) {
 		if runtime.GOOS == "darwin" || runtime.GOOS == "windows" {
 			return true
 		}
+		return IsLinuxDockerDesktop(name)
 	}
 	return false
 }
 
+// GenerateClientForLinuxDockerDesktop returns the host of docker daemon when using linux docker desktop
+func GenerateClientForLinuxDockerDesktop() (*client.Client, error) {
+	cmd := exec.Command("docker", "context", "inspect", "desktop-linux")
+	stdout, _ := cmd.StdoutPipe()
+	cmd.Start()
+	data, _ := ioutil.ReadAll(stdout)
+	var objs []interface{}
+	json.Unmarshal(data, &objs)
```

Comment:
Lots of missing error checking, we always want to make sure to check for errors to prevent `panic`ing and log all errors for debugging.